### PR TITLE
adde support for v2 api Settings

### PR DESCRIPTION
**Pull request by wouter-evolane**
added v2 settings support and tests

### CODE REVIEW

**Comment by Dynatrace-James-Kitson** [comment]
Thank you for your contributions!

Overall it looks like a good start on the settings API. There's a few small issues around formatting and consistency which we can clean up. I'll make a few notes, if you'd like to work on those yourself otherwise I can merge and make the changes myself.

File: dynatrace/environment_v2/settings.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+from datetime import datetime
+from dynatrace.dynatrace_object import DynatraceObject
+from typing import List, Optional, Union, Dict, Any
+
+from dynatrace.http_client import HttpClient
+from dynatrace.pagination import PaginatedList
+import json
+import logging
+from http.client import HTTPConnection  # py3
```
Comment (Dynatrace-James-Kitson) [inline dynatrace/environment_v2/settings.py:9]: I don't think this is used anywhere?

File: dynatrace/environment_v2/settings.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+
+class SettingService:
+    ENDPOINT = "/api/v2/settings/objects"
+
+    def __init__(self, http_client: HttpClient):
+        self.__http_client = http_client
+    
+    def list(self,schema_id: Optional[str] = None,
```
Comment (Dynatrace-James-Kitson) [inline dynatrace/environment_v2/settings.py:18]: We'll want to update these method names to 'list_objects' etc... The issue would be that this API also lets you list and view 'schemas' so we'd need to name them a bit more clearly to differentiate.

File: dynatrace/environment_v2/settings.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            "scope": scope,
+            "fields": fields,
+            "externalIds": external_ids,
+            "filter": filter,
+            "sort": sort,
+            "pageSize": page_size,
+        }
+        return PaginatedList(Settings, self.__http_client, target_url=self.ENDPOINT, list_item="items", target_params=params)
+    
+    def post(self,external_id,object_id,schema_id,schema_version,scope, value,validate_only):
```
Comment (Dynatrace-James-Kitson) [inline dynatrace/environment_v2/settings.py:37]: Similar to above the name should clarify what we're interacting with as well as we want to name them after the action not the HTTP method.

e.g. `create_object`

File: dynatrace/environment_v2/settings.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    def delete(self, object_id: str):
+        """Deletes the specified object
+
+        :param object_id: the ID of the object
+        :return: HTTP response
+        """
+        return self.__http_client.make_request(path=f"{self.ENDPOINT}/{object_id}", method="DELETE")
+
+class Settings(DynatraceObject):
+    value = None
```
Comment (Dynatrace-James-Kitson) [inline dynatrace/environment_v2/settings.py:91]: I don't think this is needed/used anywhere.

File: dynatrace/environment_v2/settings.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    def _create_from_raw_data(self, raw_element: Dict[str, Any]):
+        # Mandatory
+        self.objectId: str = raw_element["objectId"]
+        self.value: str = raw_element["value"]
+    def to_json(self):
```
Comment (Dynatrace-James-Kitson) [inline dynatrace/environment_v2/settings.py:96]: I understand the motivation for this but I don't think we'd include this as all it is is a 'getter' that returns the value attribute. If this is needed they can just access the 'value' attribute of the settings object. We don't have anything like this for other objects. to_json() could also confuse someone if they expected the _whole_ json including the object id.

**Review by Dynatrace-James-Kitson — COMMENTED** [review]
Some formatting using black also should be done (see the contributions guide doc about this).

Overall a very good start and can be merged, but will need some changes before inclusion in a release. I'll give some time if you'd like to make some more changes otherwise I can merge and do on top.

File: dynatrace/environment_v2/settings.py (hunk 5 above)
Comment (wouter-evolane) [reply dynatrace/environment_v2/settings.py]: removed to_json

File: dynatrace/environment_v2/settings.py (hunk 4 above)
Comment (wouter-evolane) [reply dynatrace/environment_v2/settings.py]: removed value=none

File: dynatrace/environment_v2/settings.py (hunk 3 above)
Comment (wouter-evolane) [reply dynatrace/environment_v2/settings.py]: renamed methods

File: dynatrace/environment_v2/settings.py (hunk 2 above)
Comment (wouter-evolane) [reply dynatrace/environment_v2/settings.py]: renamed methods

**Review by Dynatrace-James-Kitson — APPROVED** [review]

**Comment by Dynatrace-James-Kitson** [comment]
I've been adding to this commit  this week. Looking through more and testing I've made a few changes to have it align better with the rest of this project and am also deciding on some details about the 'settings object' create behavior to make it line up properly with the API while also being usable.

**Pull request merged**